### ISSUE TITLE
Option to not include variable type

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#75]
Example:

```
id obj = [array firstObject];
```

will convert to:

```
var obj: AnyObject = array.firstObject()
```

Aside from the fact that `.firstObject()` should be `.first`, it should be `var obj: AnyObject?` (notice the conditional `?`).

In most cases the type is inferred and I think that it shouldn't be included by default; or at least it should be an option to leave it out, so that the result of:

```
NSMutableArray *components = ...;
NSString *lastComponent = ...;
```

would be:

```
var components = ...
var lastComponent = ...
```

instead of

```
var components: NSMutableArray = ...
var lastComponent: String = ...
```

**Comment by drkameleon** [comment]
Hmmm...

I'll have to think about this.

**Questions:**

- How do we know that a variable is "optional" beforehand and thus add the `?`? (I mean by only having its right-hand value part)

> In most cases the type is inferred 

True. The thing is: can those "most cases" speak for... all of the cases? Wouldn't simply ignoring the type in variable declarations cause other issues? Just a thought...

**Comment by drkameleon** [comment]
Regarding `firstObject` vs `first`, I opened another issue (for keeping things clear).

However, I still have an issue. Can you please have a look here: https://github.com/drkameleon/iswift/issues/76#issuecomment-145245726

**Comment by charlieMonroe** [comment]
This can be read from the nullability information in documentation (most APIs now include `nullable` or `nonnull` attributes).

99% of cases do have the type inferred. The only place where the type isn't inferred is when `id` is returned - but then the object should likely be cast to something anyway...

IMHO you get extra errors when including the type.

**Comment by drkameleon** [comment]
> most APIs now include `nullable` or `nonnull` attributes

True. I still cannot get used to them popping up everywhere in Xcode... lol

**Comment by drkameleon** [comment]
Issue fixed as of the upcoming 1.9 - *for now, using an option in the Preferences panel*.

---

![screen shot 2015-10-03 at 5 39 17 pm](https://cloud.githubusercontent.com/assets/1265028/10263639/dabbbb44-69f5-11e5-9077-44c1999293b9.png)
